scripts/gen-service-kustomize.sh

```diff
@@ -47,6 +47,23 @@ fi
 
 pushd ${DEPLOY_DIR}
 
+STORAGE_CLASS_REPLACE=""
+
+if [ "$KIND" == "Glance" ]; then
+      STORAGE_CLASS_REPLACE="
+    - op: replace
+      path: /spec/storageClass
+      value: ${STORAGE_CLASS}"
+elif [ "$KIND" == "OpenStackControlPlane" ]; then
+      STORAGE_CLASS_REPLACE="
+    - op: replace
+      path: /spec/storageClass
+      value: ${STORAGE_CLASS}
+    - op: replace
+      path: /spec/glanceTemplate/storageClass
+      value: ${STORAGE_CLASS}"
```

Comment:
do we need to set this with https://github.com/openstack-k8s-operators/openstack-operator/blob/fb0de2a0805580ddd928cfacb3e38b44f171d035/pkg/openstack/glance.go#L36 ? if not, we could just blind set /spec/storageClass for all service CRs, if they don't have it, it gets ignored by the api server and we'd not have to do this service condition kind check here?

Reply:
just thinking about not adding service specific things here which requires updates when we add additional services.

Reply:
That's a good point actually.  Now that we have that `OpenStackControlPlane` `storageClass` propagation patch [1] merged for Glance, we can just do a blind `replace` here with Kustomize without checking anything else.  Really the `KIND` comparison was just to handle the longer paths, because Kustomize throws an error if an intermediate dir is missing in the path (i.e. if you wanted to replace `spec/glanceTemplate/storageClass` and the CRD did not have `glanceTemplate` in the spec).  But if we're always doing `/spec/storageClass` we should be fine.

[1] https://github.com/openstack-k8s-operators/openstack-operator/pull/29

Reply:
Changed to use the blind `replace`